File: blog/src/main/kotlin/com/liux/blog/bean/po/Link.kt
package com.liux.blog.bean.po

import com.liux.blog.bean.NoArgsConstructor
import java.util.*

// 友链数据
@NoArgsConstructor
data class Link(
    // ID
    var id: Int? = null,
    // 标题
    var title: String? = null,
    // 地址
    var url: String? = null,
    // 权重
    var weight: Int? = null,
    // 状态 -1_已删除 0_未启用 1_已启用
    var status: Int? = null,
    // 创建时间
    var createTime: Date? = null,
    // 更新时间
    var updateTime: Date? = null,
)

File: blog/bean/api_vo/link.go
package api_vo

import (
	"blog/bean/po"
	"time"
)

type LinkVO struct {
	Id         int        `json:"id"`
	Title      string     `json:"title"`
	Url        string     `json:"url"`
	Weight     int        `json:"weight"`
	Status     int        `json:"status"`
	CreateTime time.Time  `json:"createTime"`
	UpdateTime *time.Time `json:"updateTime"`
}

func (v *LinkVO) From(link po.Link) {
	v.Id = link.Id
	v.Title = link.Title
	v.Url = link.Url
	v.Weight = link.Weight
	v.Status = link.Status
	v.CreateTime = link.CreateTime
	v.UpdateTime = link.UpdateTime
}
